utilities/derives/src/test_with_metrics.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use proc_macro2::{Ident, Span};
+use proc_macro_crate::crate_name;
+use quote::quote;
+use syn::ItemFn;
+
+pub(crate) fn generate_test_function(function: ItemFn) -> proc_macro::TokenStream {
```

Comment:
@ljedrz @niklaslong Do we know why this is here? I think we should remove it. If `snarkOS` requires it, seems like it should be moved there.

Reply:
I don't recall this, and agree that any snarkos-specific functionality should be implemented there.

Reply:
I don't recall either 🤔